addons/project_timesheet_holidays/models/res_config_settings.py

```diff
@@ -1,11 +1,16 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
-    leave_timesheet_project_id = fields.Many2one(related='company_id.leave_timesheet_project_id', string="Internal Project", domain="[('company_id', '=', company_id)]", readonly=False)
+    leave_timesheet_project_id = fields.Many2one(related='company_id.leave_timesheet_project_id', required=True, string="Internal Project", domain="[('company_id', '=', company_id)]", readonly=False)
```

Comment:
Seems louche to have a field readonly and required. Put it in the view only instead of in the model. 

The risk here is that it could be impossible to save the record in that case.